### ISSUE TITLE
Order of paragraphs and tables

### ISSUE THREAD

**Issue by Prigin** [ruby-docx/docx#118]
## Problem

I need to get all paragraphs and tables in order they have in docx file. Is there any way I can do this?

## Solution

May be just one index for paragraph objects and table objects will be enough.

**Comment by satoryu** [comment]
You mean that at the latest version of this gem `Document#paragraphs` returns paragraphs in wrong order, right?
Could you give us a docx file to reproduce this behavior if you have? 
The file would help us to investigate what happens. 

Thanks

**Comment by Prigin** [comment]
Not exactly. :) Sorry for not being transparent. Lets say I have a docx:
`DOCX:
   1. paragraph1
   2. paragraph2
   3. table1
   4. paragraph4
   5. paragraph5
   6. table2
`
How to get the same order of elements they have in DOCX? Or maybe they already have that method(which returns order number from doc). I cant actually find it :(

**Comment by aunghtain** [comment]
I was able to do this as followed. I'm using private vars/methods, but if they open up more APIs in the future, we won't have to.
`
    doc = Docx::Document.open(file)
    doc.instance_variable_get("@doc").xpath('//w:document//w:body').children.each do |c|
      if c.name == 'p' # paragraph
        p = doc.send(:parse_paragraph_from, c)      
      elsif c.name = 'tbl' # table
        t = doc.send(:parse_table_from, c)  
      else # other types?
      end
    end

`

**Comment by aunghtain** [comment]
if u just want text, u don't need to parse them as paragraph/table. u can just get as "c.content"
